fl4health/feature_alignment/tabular_feature.py

```diff
@@ -0,0 +1,73 @@
+import json
+from typing import Optional, Union
+
+from flwr.common.typing import Dict, List, Scalar
+
+from fl4health.feature_alignment.tabular_type import TabularType
+
+MetaData = Union[Dict[str, int], List[Scalar]]
+
+
+class TabularFeature:
+    def __init__(
+        self, feature_name: str, feature_type: TabularType, fill_value: Optional[Scalar], metadata: MetaData = []
+    ) -> None:
+        """
+        Information that represents a tabular feature.
+
+        Args:
+            feature_name (str): name of the feature.
+            feature_type (TabularType): data type of the feature.
+            fill_value (Optional[Scalar]): the default fill value for this feature when it is missing in a dataframe.
+            metadata (MetaData, optional): metadata associated with this feature.
+            For example, if the feature is categorical, then metadata would be all the categories. Defaults to [].
+        """
+        self.feature_name = feature_name
+        self.feature_type = feature_type
+        # Each TabularType has its own default fill value, which is used
+        # when the feature does not have its default fill value specified.
+        if fill_value is None:
+            self.fill_value = TabularType.get_default_fill_value(self.feature_type)
+        else:
+            self.fill_value = fill_value
+        self.metadata = metadata
+
+    def get_feature_name(self) -> str:
+        return self.feature_name
+
+    def get_feature_type(self) -> TabularType:
+        return self.feature_type
+
+    def get_fill_value(self) -> Scalar:
+        return self.fill_value
+
+    def get_metadata(self) -> MetaData:
+        return self.metadata
+
+    def get_metadata_dimension(self) -> int:
+        if self.feature_type == TabularType.BINARY or TabularType.ORDINAL:
+            return len(self.metadata)
+        elif self.feature_type == TabularType.NUMERIC:
+            return 1
+        else:
```

Comment:
Rather than returning zero here, does it make sense to just raise an exception, as getting the dimension for other feature types isn't "supported"? Maybe there are downstream implications though, so correct me if I'm wrong.

Reply:
It's true we are assuming that the target feature cannot be of type string, but since this method is for the class `TabularFeature` I think it's kind of odd that it would raise an error only for certain instances of the class but not others. Originally I thought to return the length of the vocabulary for string features here, but because string pipelines produce sparse matrices, such a length would not reflect the actual "dimension" of the output. So I just put zero here instead... I don't think there is any downstream implication though, so let me know what you think is the best.

Reply:
I would say that it's okay for us to raise an exception for only some `feature_types` here if using `get_metadata_dimension` on them is "improper" and should be avoided. Basically the exception would represent that something weird has happened that we're calling `get_metadata_dimension` on a feature type that doesn't really have one. However, I may be missing something.

Reply:
Sounds good!